orders/views.py

```diff
@@ -121,4 +124,86 @@ def patch(self, request, cart_item):
             return JsonResponse({'message':'SUCCESS'}, status=200)
 
         except KeyError:
-            return JsonResponse({'message':'KEY_ERROR'}, status=400)
\ No newline at end of file
+            return JsonResponse({'message':'KEY_ERROR'}, status=400)
+
+class PurchaseView(View):
+    @login_decorator
+    @transaction.atomic
+    def post(self, request):
+        try:
+            data           = json.loads(request.body)
+            signed_user    = request.user
+            cart_items     = CartItem.objects.filter(user=signed_user).select_related('product_options', 'product_options__product')
+            total_quantity = 0
+            total_price    = 0
+            coupon_id      = data.get('couponId', None)
+
+            if not cart_items:
+                return JsonResponse({"message":"NO_ITEMS_IN_CART"}, status=400)
```

Comment:
```suggestion
            if not cart_items.exists():
                return JsonResponse({"message":"NO_ITEMS_IN_CART"}, status=400)
```
존재 유무를 판단 할 때는 exists()을 이용해 주세요. 안 그러면 모든 데이터를 다 가져온 뒤에 존재 유무를 판단하게 됩니다.

Reply:
넵 수정했습니다